search-service/src/main/java/de/cxp/ocs/elasticsearch/query/FiltersBuilder.java

```diff
@@ -82,24 +107,12 @@ public FilterContext buildFilterContext(List<InternalResultFilter> filters) {
 				}
 			}
 
-			if (isBasicQuery(filter.getField().getName())) {
+			if (isBasicQuery(filter.getField().getName()) || addAllFiltersAsBasicFilters) {
 				basicFilterQueries.put(filter.getField().getName(), filterQuery);
-			}
-			else {
+			} else {
```

Comment:
Ok, got it: it works, because the most incoming parameters come from multi-select facets. In such case, the user-selection will be executed as post-filter and the querqy-filter will be executed as basic filter.
However if we have a single-select facet, other filters for that field will be overwritten. This is fine unless, that filtered products also have other possible values on the same field/attribute for the remaining products...

Reply:
The fix for that is fairly easy, although not optimal, since the resulting ES query becomes a real monster. However it works.